modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathLocator.cs

```diff
@@ -1,185 +1,293 @@
 {
+    internal enum OperatingSystemFamily
+    {
+        /// <summary>
+        ///   <para>Returned for operating systems that do not fall into any other category.</para>
+        /// </summary>
+        Other,
+
+        /// <summary>
+        ///   <para>macOS operating system family.</para>
+        /// </summary>
+        MacOSX,
+
+        /// <summary>
+        ///   <para>Windows operating system family.</para>
+        /// </summary>
+        Windows,
+
+        /// <summary>
+        ///   <para>Linux operating system family.</para>
+        /// </summary>
+        Linux
+    }
+
+    internal interface IRiderLocatorEnvironment
+    {
+        OperatingSystemFamily operatingSystemFamily { get; }
+
+        T FromJson<T>(string json);
+        void Warn(string message, Exception e = null);
+        void LogException(Exception exception);
+    }
+
     /// <summary>
     /// This code is a modified version of the JetBrains resharper-unity plugin listed under Apache License 2.0 license:
     /// https://github.com/JetBrains/resharper-unity/blob/master/unity/JetBrains.Rider.Unity.Editor/EditorPlugin/RiderPathLocator.cs
     /// </summary>
-    public static class RiderPathLocator
+    internal class RiderPathLocator
     {
-        public static RiderInfo[] GetAllRiderPaths()
+        public readonly IRiderLocatorEnvironment RiderLocatorEnvironment;
+
+        internal RiderPathLocator(IRiderLocatorEnvironment riderLocatorEnvironment)
+        {
+            RiderLocatorEnvironment = riderLocatorEnvironment;
+        }
+
+        [UsedImplicitly] // Used in com.unity.ide.rider
+        public RiderInfo[] GetAllRiderPaths()
         {
             try
             {
-                if (OS.IsWindows)
-                {
-                    return CollectRiderInfosWindows();
-                }
-                if (OS.IsMacOS)
+                switch (RiderLocatorEnvironment.operatingSystemFamily)
                 {
-                    return CollectRiderInfosMac();
+                    case OperatingSystemFamily.Windows:
+                        return CollectRiderInfosWindows();
+                    case OperatingSystemFamily.MacOSX:
+                        return CollectRiderInfosMac();
+                    case OperatingSystemFamily.Linux:
+                        return CollectAllRiderPathsLinux();
                 }
-                if (OS.IsUnixLike)
-                {
-                    return CollectAllRiderPathsLinux();
-                }
-                throw new InvalidOperationException("Unexpected OS.");
             }
             catch (Exception e)
             {
-                GD.PushWarning(e.Message);
+                RiderLocatorEnvironment.LogException(e);
             }
 
-            return Array.Empty<RiderInfo>();
+            return new RiderInfo[0];
         }
 
-        private static RiderInfo[] CollectAllRiderPathsLinux()
+        private RiderInfo[] CollectAllRiderPathsLinux()
```

Comment:
All of these static methods are now instance methods but that doesn't seem like a necessary change.